src/main/com/cqfn/degitx/simulator/Node.kt

```diff
@@ -14,17 +12,42 @@ class Node(override var hardware: Hardware,
 
     constructor(hw: Hardware): this(hw, State.ACTIVE)
 
-    override var rqPool: Queue<Request> = LinkedList<Request>()
-
     /**
      * Logic of software GitHub Node
+     * If current state of the Node allows push Request to process queue.
      */
-    override fun run(addr: Address, rq: Request) {
-        // Process incoming requests to THIS node
-        var req = rqPool.poll()
-        while (req != null) {
-           // req.
+    override fun run(rq: Request) {
+        if (this.state == State.ACTIVE) {
+            this.hardware.net.rqPool.add(rq)
+            processRequest()
+        } else {
+            println("Node with address " + this.hardware.net.addr.addr +
+                    " cannot process request as it is " + this.state.toString())
         }
     }
 
+    /**
+     * Node internal logic for Request processing
+     */
+    fun processRequest() {
+        var rq : Request = this.hardware.net.receive().poll()
+        while (rq != null) {
+            if (rq.addressee.equals(this.hardware.net.addr)) {
+                // Process incoming requests to THIS node
```

Comment:
@g4s8 do we allows comments inside a method? Or do we follow [qulice](https://www.qulice.com/) approach and disallow comments inside a method and force developers to extract methods in such cases?

Reply:
@stepanvalyavskiy ,Yes, you're right. I'll move it to main javadoc.